### PR TITLE
Move extensions to top-level functions.

### PR DESCRIPTION
**Pull request by JakeWharton**
IntelliJ IDEA cannot currently discover them when embedded inside an companion object. Plus this makes better imports in the consuming code.

Refs #157.

### CODE REVIEW

File: src/main/java/com/squareup/kotlinpoet/WildcardTypeName.kt [hunk 1]
```diff
@@ -121,3 +118,9 @@ class WildcardTypeName private constructor(
     }
   }
 }
+
+@JvmName("get")
+fun javax.lang.model.type.WildcardType.asWildcardTypeName() = Companion.get(this, mutableMapOf())
```
Comment (Egorand) [inline src/main/java/com/squareup/kotlinpoet/WildcardTypeName.kt]: Why not `WildcardTypeName.get(this, mutableMapOf())`?

Comment (JakeWharton) [inline src/main/java/com/squareup/kotlinpoet/WildcardTypeName.kt]: Probably just the result of automated refactoring. Will fix.

File: src/main/java/com/squareup/kotlinpoet/NameAllocator.kt [hunk 2]
```diff
@@ -84,7 +84,7 @@ class NameAllocator private constructor(
   /**
    * Return a new name using `suggestion` that will not be a Java identifier or clash with other
    * names. The returned value can be queried multiple times by passing `tag` to
-   * [NameAllocator.get].
+   * [NameAllocator.fromType].
```
Comment (NightlyNexus) [inline src/main/java/com/squareup/kotlinpoet/NameAllocator.kt]: it's still `get`. accidental change?

File: src/main/java/com/squareup/kotlinpoet/TypeName.kt [hunk 3]
```diff
@@ -51,7 +52,7 @@ import kotlin.reflect.KClass
  * --------------------------
  *
  * In an annotation processor you can get a type name instance for a type mirror by calling
- * [TypeName.get]. In reflection code, you can use [TypeName.get].
+ * [TypeName.fromTypeMirror]. In reflection code, you can use [TypeName.fromType].
```
Comment (NightlyNexus) [inline src/main/java/com/squareup/kotlinpoet/TypeName.kt]: the static method names are still all `get`.

File: src/main/java/com/squareup/kotlinpoet/NameAllocator.kt (hunk 2 above)
Comment (JakeWharton) [inline src/main/java/com/squareup/kotlinpoet/NameAllocator.kt]: Just IntelliJ being stupid when it's trying to be really smart.

File: src/main/java/com/squareup/kotlinpoet/WildcardTypeName.kt (hunk 1 above)
Comment (Egorand) [inline src/main/java/com/squareup/kotlinpoet/WildcardTypeName.kt]: Cool, that `Companion` import looked weird

File: src/main/java/com/squareup/kotlinpoet/TypeName.kt (hunk 3 above)
Comment (JakeWharton) [inline src/main/java/com/squareup/kotlinpoet/TypeName.kt]: They're `get` in Java but `asTypeName` in Kotlin

**Pull request merged**